control-plane/subcommand/gateway-resources/command.go

```diff
@@ -331,37 +359,104 @@ func (c *Command) loadConfig() error {
 	return nil
 }
 
+// loadMeshConfigs reads and loads the mesh configs from `/consul/config/config.yaml`, if this file does not exist nothing is done
+func (c *Command) loadMeshGatewayConfigs() error {
+	file, err := os.Open(c.flagMeshGatewayConfigLocation)
+	if err != nil {
+		if os.IsNotExist(err) {
+			c.UI.Warn("mesh gateway configuration file not found, skipping mesh gateway configuration")
+			return nil
+		}
+		c.UI.Error(fmt.Sprintf("Unable to open mesh configuration config.yaml: %s", err))
+		return err
+	}
+
+	config, err := io.ReadAll(file)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Unable to read mesh configuration config.yaml: %s", err))
+		return err
+	}
+
+	err = k8syaml.Unmarshal(config, &c.meshGatewayConfig)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error decoding mesh gateway config: %s", err))
+		return err
+	}
+
+	return nil
+}
+
+// createMeshGatewayClassAndClassConfigs utilizes the configuration loaded from the mesh gateway config file to
+// create the GatewayClassConfig and GatewayClass for the mesh gateway
+func (c *Command) createMeshGatewayClassAndClassConfigs(ctx context.Context) error {
+	labels := map[string]string{
+		"app":       c.flagApp,
+		"chart":     c.flagChart,
+		"heritage":  c.flagHeritage,
+		"release":   c.flagRelease,
+		"component": "mesh-gateway",
+	}
+	for _, cfg := range c.meshGatewayConfig.GatewayClassConfigs {
+		err := forceV2ClassConfig(ctx, c.k8sClient, cfg)
+		if err != nil {
+			return err
+		}
+
+		class := &v2beta1.GatewayClass{
+			ObjectMeta: metav1.ObjectMeta{Name: cfg.Name, Labels: labels},
+			Spec: meshv2beta1.GatewayClass{
+				ControllerName: "mesh-gateway-controller",
+				ParametersRef: &meshv2beta1.ParametersReference{
+					Group:     "mesh.consul.hashicorp.com",
+					Kind:      "gatewayClass",
+					Namespace: &cfg.Namespace,
+					Name:      cfg.Name,
+				},
+			},
+		}
+
+		err = forceV2Class(ctx, c.k8sClient, class)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 func (c *Command) Synopsis() string { return synopsis }
 func (c *Command) Help() string {
 	c.once.Do(c.init)
 	return c.help
 }
 
-const synopsis = "Create managed gateway resources after installation/upgrade."
-const help = `
+const (
+	synopsis = "Create managed gateway resources after installation/upgrade."
+	help     = `
 Usage: consul-k8s-control-plane gateway-resources [options]
 
   Installs managed gateway class and configuration resources
 	after a helm installation or upgrade in order to avoid the
 	dependencies of CRDs being in-place prior to resource creation.
 
 `
+)
 
-func defaultResourceRequirements() v1.ResourceRequirements {
+func defaultResourceRequirements() corev1.ResourceRequirements {
```

Comment:
side effect of this change https://github.com/hashicorp/consul-k8s/pull/3273/files#r1406708181